### PR TITLE
feat: added support for date translations

### PR DESCRIPTION
**Pull request by wadeking98**
# Summary of Changes

Added support for month translation in bifold dates.
Added exact issued time to credential detail screen.
Updated jest tests

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/screens/ContactDetails.tsx [hunk 1]
```diff
@@ -154,7 +154,7 @@ const ContactDetails: React.FC<ContactDetailsProps> = ({ route }) => {
           {
             name: connection?.alias || connection?.theirLabel,
             value: t('ContactDetails.DateOfConnection', {
-              date: connection?.createdAt.toLocaleString(i18n.language, dateFormatOptions),
+              date: connection?.createdAt !== undefined ? formatTime(connection.createdAt) : '',
```
Comment (amanji) [inline core/App/screens/ContactDetails.tsx]: Just FYI: The inclusion of the optional chaining (`?.`) operator in `connection?.createdAt` is the same thing as `connection.createdAt !== undefined`. So you can simply write `connection?.createdAt ? formatTime(connection.createdAt) : ''`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: core/App/screens/ProofRequestAttributeDetails.tsx [hunk 2]
```diff
@@ -149,8 +154,7 @@ const ProofRequestAttributeDetails: React.FC<ProofRequestAttributeDetailsProps>
               </View>
             ) : (
               <Text style={ListItems.recordAttributeText} testID={testIdWithKey('Issued')}>
-                {t('CredentialDetails.Issued')}{' '}
-                {credential.createdAt.toLocaleDateString(i18n.language, dateFormatOptions)}
+                {t('CredentialDetails.Issued')} {formatTime(credential.createdAt)}
```
Comment (amanji) [inline core/App/screens/ProofRequestAttributeDetails.tsx]: Space removed?

File: core/App/utils/helpers.ts [hunk 3]
```diff
@@ -29,6 +31,54 @@ export function hashToRGBA(i: number) {
   return '#' + '00000'.substring(0, 6 - colour.length) + colour
 }
 
+export function formatTime(time: Date, params?: { long?: boolean; format?: string }): string {
+  const getMonthKey = 'MMMM'
+  const momentTime = moment(time)
+  const monthKey = momentTime.format(getMonthKey)
+  const customMonthFormatRe = /M+/
+  const long = params?.long
```
Comment (amanji) [inline core/App/utils/helpers.ts:74]: You can shorten this to `const { long, format } = params`

File: core/App/utils/helpers.ts [hunk 4]
```diff
@@ -29,6 +31,54 @@ export function hashToRGBA(i: number) {
   return '#' + '00000'.substring(0, 6 - colour.length) + colour
 }
 
+export function formatTime(time: Date, params?: { long?: boolean; format?: string }): string {
```
Comment (amanji) [inline core/App/utils/helpers.ts:69]: Question: Do locales not work in moment.js?

Comment (wadeking98) [reply core/App/utils/helpers.ts:69]: no, it seems locales don't do anything in react-native period. They don't change the translation and they don't change the date format

File: core/App/utils/helpers.ts (hunk 3 above)
Comment (wadeking98) [reply core/App/utils/helpers.ts:74]: I tried that but then I get this error for each property: `Property 'format' does not exist on type '{ long?: boolean | undefined; format?: string | undefined; } | undefined'`

File: core/App/screens/CredentialDetails.tsx [hunk 5]
```diff
@@ -185,6 +184,12 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
                 <Text style={[TextTheme.title]}>{t('CredentialDetails.IssuedBy')}</Text>{' '}
                 <Text style={[TextTheme.normal]}>{credentialConnectionLabel}</Text>
               </Text>
+              <Text>
+                <Text style={[TextTheme.title]}>{t('CredentialDetails.Issued')}:</Text>{' '}
+                <Text style={[TextTheme.normal]}>
+                  {credential?.createdAt !== undefined && formatTime(credential?.createdAt, { long: true })}
```
Comment (amanji) [inline core/App/screens/CredentialDetails.tsx]: Same comment above about using optional chaining

File: core/App/screens/CredentialDetails.tsx [hunk 6]
```diff
@@ -185,6 +184,12 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
                 <Text style={[TextTheme.title]}>{t('CredentialDetails.IssuedBy')}</Text>{' '}
                 <Text style={[TextTheme.normal]}>{credentialConnectionLabel}</Text>
               </Text>
+              <Text>
+                <Text style={[TextTheme.title]}>{t('CredentialDetails.Issued')}:</Text>{' '}
```
Comment (amanji) [inline core/App/screens/CredentialDetails.tsx]: I like the format you used in another file:

`{t('CredentialDetails.Issued') + ' '} {formatTime(credential.createdAt)}`

Could probably even shorten it to:

<pre>`{${t('CredentialDetails.Issued')} ${formatTime(credential.createdAt)}}`</pre>

File: core/App/screens/ProofRequestAttributeDetails.tsx [hunk 7]
```diff
@@ -149,8 +154,7 @@ const ProofRequestAttributeDetails: React.FC<ProofRequestAttributeDetailsProps>
               </View>
             ) : (
               <Text style={ListItems.recordAttributeText} testID={testIdWithKey('Issued')}>
-                {t('CredentialDetails.Issued')}{' '}
-                {credential.createdAt.toLocaleDateString(i18n.language, dateFormatOptions)}
+                {t('CredentialDetails.Issued') + ' '} {formatTime(credential.createdAt)}
```
Comment (amanji) [inline core/App/screens/ProofRequestAttributeDetails.tsx:181]: See comment above

File: core/App/utils/helpers.ts (hunk 4 above)
Comment (amanji) [reply core/App/utils/helpers.ts:69]: 😢

File: core/App/utils/helpers.ts (hunk 3 above)
Comment (amanji) [reply core/App/utils/helpers.ts:74]: Ah ok, I guess because `params` is optional itself then it can be `undefined`.

**Review by amanji — DISMISSED** [review]

**Review by jleach — APPROVED** [review]
